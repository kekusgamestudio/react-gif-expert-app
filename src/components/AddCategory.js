import React, { useState } from 'react'
import PropTypes from 'prop-types';


const AddCategory = ({ setCategoriesFunc }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        //console.log(e.target.value);
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            if (inputValue.substring(0, 4) === 'NEW:') {
                setCategoriesFunc((cats) => [inputValue.substring(4).trim()]);
            } else {
                setCategoriesFunc((cats) => [inputValue, ...cats]);
            }

            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p> {inputValue} </p>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}


export default AddCategory
