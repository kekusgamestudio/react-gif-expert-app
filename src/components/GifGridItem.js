import React from 'react';
import PropTypes from 'prop-types';
;


export const GifGridItem = ({ title, url }) => {
    //console.log(url)

    return (
        <div className="card animate__animated animate__flipInY">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}