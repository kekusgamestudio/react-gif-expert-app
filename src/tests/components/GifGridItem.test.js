import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';
import { shallow } from 'enzyme';


describe('Pruebas en <GifGridItem />', () => {

    const title = 'Un t√≠tulo';
    const url = 'https://localhost/algo.jpg';

    test('Debe mostrar <GifGridItem /> correctamente', () => {
        const wrapper = shallow(<GifGridItem title={title} url={url} />)
        expect(wrapper).toMatchSnapshot();
    });

});


