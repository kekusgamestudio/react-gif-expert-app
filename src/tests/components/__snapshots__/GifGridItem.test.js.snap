// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__flipInY"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
